src/indra_cogex/client/queries.py

```diff
@@ -1740,6 +1794,1447 @@ def _filter_out_medscan_evidence(
+
+    Parameters
+    ----------
+    client : Neo4jClient
+        The Neo4j client
+    phenotype : Tuple[str, str]
+        The phenotype to query (e.g., ("mesh", "D001827"))
+
+    Returns
+    -------
+    :
+        Variant nodes (DBSNP) associated with this phenotype
+    """
+    return client.get_sources(
+        phenotype,
+        relation="variant_phenotype_association",
+        source_type="BioEntity",
+        target_type="BioEntity"
+    )
+
+
+@autoclient()
+def has_variant_phenotype_association(
+    variant: Tuple[str, str], phenotype: Tuple[str, str], *, client: Neo4jClient
+) -> bool:
+    """Check if a variant is associated with a phenotype in GWAS data.
+
+    Parameters
+    ----------
+    client : Neo4jClient
+        The Neo4j client
+    variant : Tuple[str, str]
+        The variant to query (e.g., ("dbsnp", "rs13015548"))
+    phenotype : Tuple[str, str]
+        The phenotype to query (e.g., ("mesh", "D001827"))
+
+    Returns
+    -------
+    :
+        True if the variant is associated with the phenotype
+    """
+    return client.has_relation(
+        variant,
+        phenotype,
+        relation="variant_phenotype_association",
+        source_type="BioEntity",
+        target_type="BioEntity"
+    )
+
+
+# chembl
+@autoclient()
+def get_indications_for_molecule(
+    molecule: Tuple[str, str], *, client: Neo4jClient
+) -> Iterable[Node]:
+    """Return indications associated with the given molecule from ChEMBL.
+
+    Parameters
+    ----------
+    client : Neo4jClient
+        The Neo4j client
+    molecule : Tuple[str, str]
+        The molecule to query (e.g., ("chebi", "10001"))
+
+    Returns
+    -------
+    :
+        Disease nodes (MESH) associated with this molecule
+    """
+    molecule_id = f"{molecule[0]}:{molecule[1]}".lower()
+    query = """
+    MATCH (m:BioEntity {id: $molecule_id})-[r:has_indication]->(i:BioEntity)
+    RETURN i
+    """
+    return client.query_nodes(query, molecule_id=molecule_id)
+
+
+@autoclient()
+def get_molecules_for_indication(
+    indication: Tuple[str, str], *, client: Neo4jClient
+) -> Iterable[Node]:
+    """Return molecules associated with the given indication from ChEMBL.
+
+    Parameters
+    ----------
+    client : Neo4jClient
+        The Neo4j client
+    indication : Tuple[str, str]
+        The disease indication to query (e.g., ("mesh", "D002318"))
+
+    Returns
+    -------
+    :
+        Molecule nodes (CHEBI or CHEMBL) associated with this indication
+    """
+    indication_id = f"{indication[0]}:{indication[1]}" if indication[
+                                                              0].lower() == "mesh" else f"{indication[0]}:{indication[1]}".lower()
+    query = """
+    MATCH (m:BioEntity)-[r:has_indication]->(i:BioEntity {id: $indication_id})
+    RETURN m
+    """
+    return client.query_nodes(query, indication_id=indication_id)
+
+
+@autoclient()
+def has_indication(
```

Comment:
We could use a better name here too, maybe `molecule_has_indidcation`.

Reply:
Fixed in 4640b451